### ISSUE TITLE
Compact: Invalid magic number (help)

### ISSUE THREAD

**Issue by jsedy7** [improbable-eng/thanos#956]
**Thanos, Prometheus and Golang version used**

Thanos 0.3.2, Prometheus 2.7.1

```
thanos, version 0.3.2 (branch: HEAD, revision: 4b7320c0e45e3f48a437bd19294f569785bafb02)
  build user:       root@e9a9c28f966a
  build date:       20190304-17:11:05
  go version:       go1.11.5
```

**What happened**

Thanos Compact crash during the compaction on invalid magic number. When I remove folder in S3 storage and run Compact again, It works correctly at least into found the next invalid folder. I have a lot of folders in S3 and I can't remove any folders with every single job.
Have you the best practise to work with invalid data from thanos in S3? I would like to repair it or simple delete all invalid files. I spent a lot of time with the deleting invalid folders and I think, that isn't best way. 

I started job (in k8s) - compact found invalid data - I removed invalid data - run compact again - compact found new invalid data - I removed data again ... and again and again.. 

Sometimes my job crashed on _index out of range_, but invalid magic number I see every very often.

**What you expected to happen**
Compact running for long time without the crashes. 

**How to reproduce it (as minimally and precisely as possible)**:
I can reproduce it again with next jobs. 

**Full logs to relevant components**

<details>Logs
<p>

```
level=debug ts=2019-03-21T11:53:46.564772731Z caller=compact.go:176 msg="download meta" block=01D6G134P7V4E7W8GX4620QXVZ
level=debug ts=2019-03-21T11:53:46.624741852Z caller=compact.go:194 msg="block is too fresh for now" block=01D6G134P7V4E7W8GX4620QXVZ
level=info ts=2019-03-21T11:53:46.625116193Z caller=compact.go:827 msg="start of GC"
level=debug ts=2019-03-21T11:56:15.982571369Z caller=compact.go:721 compactionGroup="0@{replica=\"exporters_NG2\"}" msg="downloaded and verified blocks" blocks="[/www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D632H4XZYQZTPFAMTD0HA96H /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D639CW608D0ZJKXF69T9XAHV /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D63G8KE29WB42SFHWMX15WV6 /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D63Q4AP21CHD26HDFD28B2HD]" duration=2m29.212187259s
level=error ts=2019-03-21T11:56:16.195438135Z caller=main.go:181 msg="running command failed" err="compaction failed: compaction: compact blocks [/www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D632H4XZYQZTPFAMTD0HA96H /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D639CW608D0ZJKXF69T9XAHV /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D63G8KE29WB42SFHWMX15WV6 /www/thanos/store/compact/0@{replica=\"exporters_NG2\"}/01D63Q4AP21CHD26HDFD28B2HD]: invalid magic number 5000403"
```

</p>
</details>

**Comment by FUSAKLA** [comment]
Hi, the `magic number` refers to the structure of series file in the block in TSDB which Prometheus and Thanos uses

https://github.com/prometheus/tsdb/blob/3ab5f4e57902eea758f1940dd852e954b5d5ba6b/chunks/chunks.go#L34

Your blocks are probably malformed somehow? What object storage are you using? Is it actual S3 or Minio for example?

To check your blocks use the `thanos bucket inspect` and `thanos bucket verify` commands see the docs 
https://github.com/improbable-eng/thanos/blob/master/docs/components/bucket.md#verify

Those should tell you which blocks are affected but the main thing is to find out what causes corruption of the blocks. Are you sure that the comaction process does not run simultaneously?

**Comment by FUSAKLA** [comment]
Hi, does the issue persist?

**Comment by jsedy7** [comment]
Hi, thank you for your response. This issue can be closed. 

Description:
I used Ceph as S3 storage for Thanos. I had a problem with radosgw restarting on OOM exception. Data from sidecars couldn't be uploaded completely. I think that thanos-compact crashed on incomplete data. This problem I had in one of datacenter, but second worked correctly. 

Fix:
- I increased resources in radosgw 
- I removed wrong files
